### DISCUSSION TITLE
map files weren't updated automatically

### DISCUSSION THREAD

**Discussion by weifengding** [martinschaible/rspamd-rules#30] [General]
Hello,

I manually downloaded all the map files and saved on my rspamd server under /etc/rspamd/maps.d folder. Those map files work fine, however, found that they never get updated since installed them on Oct 20.

/etc/rspamd/local.d/options.conf
map_watch_interval = 60min;

rspamd log:

2023-10-27 16:44:41 #397123(controller) <sdp9gz>; map; http_map_finish: data is not modified for server raw.githubusercontent.com, next check at Fri, 27 Oct 2023 23:50:56 GMT (http cache based: Fri, 27 Oct 2023 23:50:56 GMT)
2023-10-27 16:44:41 #397123(controller) <hbqi58>; map; http_map_finish: data is not modified for server raw.githubusercontent.com, next check at Fri, 27 Oct 2023 23:50:56 GMT (http cache based: Fri, 27 Oct 2023 23:50:56 GMT)
2023-10-27 16:44:45 #397123(controller) <kuwuur>; map; http_map_finish: data is not modified for server raw.githubusercontent.com, next check at Fri, 27 Oct 2023 23:51:01 GMT (http cache based: Fri, 27 Oct 2023 23:51:01 GMT)

I tried change the file ownership from root to _rspamd but it didn't help.

-rw-r--r-- 1 _rspamd _rspamd 9.9K Oct 20 09:12 _sale_en.map
-rw-r--r-- 1 _rspamd _rspamd 2.2K Oct 20 09:12 _sale_sender_de.map
-rw-r--r-- 1 _rspamd _rspamd 3.3K Oct 20 09:12 _sale_sender_en.map
-rw-r--r-- 1 _rspamd _rspamd 1.7K Oct 20 09:12 _sale_subject_de.map
-rw-r--r-- 1 _rspamd _rspamd 1.4K Oct 20 09:12 _sale_subject_en.map
-rw-r--r-- 1 _rspamd _rspamd 6.4K Oct 20 09:12 _scam_de.map
-rw-r--r-- 1 _rspamd _rspamd 6.0K Oct 20 09:12 _scam_donation_de.map

**Comment by martinschaible** [comment]
Hello

The settings `map `  in the config files in the folder `local.d` are pointing to github . Sample:

`map = "https://raw.githubusercontent.com/martinschaible/rspamd-rules/main/maps.d/_ad_agency_at.map";`

I have no idea, how Rspamd can be configured to download the files from the source periodically. The messages you receive is usually okay. This is a confirmation that some source fles did not change and therefore an update is not needed. 

I think, that your system behaves completely normal and gets the files from github. I would delete the setting `map_watch_interval`. The default value is 5 minutes, which is good.

**Reply by weifengding** [reply]
Hello Martin,

Thank you so much. I downloaded all the maps to /etc/rspamd/maps.d and thought rspamd is using those local maps files, but it actually periodically check the github URLs and use those remote maps instead. I was monitoring our rspamd history for the past 10 days, those maps are working very good.

**Comment by martinschaible** [comment]
Hello

I'm glad, that the rules are working good. Thanks!
